### ISSUE TITLE
Executing a SQL statement produces no results

### ISSUE THREAD

**Issue by jjcapron** [microsoft/vscode-mssql#871]
- MSSQL Extension Version:  1.0.0
- VSCode Version: 1.12.1
- OS Version: Windows 10

Steps to Reproduce:

1. Created profile to already established database.
2. Open new editor.
3. Pressed **F1** and then selected **MS SQL: Connect** and selected created profile.
4. Typed in **Select** statement of a table I knew was small (around 7 rows). (During the typing, intellisense suggests Table name for me therefore it is reading the schema.)
5. Pressed **F1** and then selected **MS SQL: Execute query**.
Attached screenshot shows results:
![visual studio code ms sql extension query results](https://cloud.githubusercontent.com/assets/12276127/25963673/5a55a150-363f-11e7-8cc6-7af9289fd501.jpg)
The Executing query... icon continues to turn and VS Code is not hanging.

**Comment by kevcunnane** [comment]
@jjcapron is this reliably happening for you? I can't reproduce this and it's a fairly key scenario so we'd expect to see more reports if this was failing.

**Comment by jjcapron** [comment]
Yes. At one point, when I first installed Visual Studio Code and this extension, i got it to work for a simple query. That was the only time though. I was wondering if maybe a setting was blank when it is not supposed to be but it is not giving an error. I can connect to the databases using SSMS but not with VS Code. It is Integrated Security so no need for password. Any log other than the output pane that I could view any issues?

**Comment by ehdevries** [comment]
The scenario that @jjcapron is describing has been happening to me for months, as well. Similarly:

- Windows 10 Pro version 1703
- VS Code 1.61.1 (and several earlier versions)
- MSSQL extension 1.2.0 (and several earlier versions)

 I experience the same behavior: the output console shows that the query was successful, but the results pane shows "executing query" spinning ad infinitum. It worked for a while, then it broke for a while, then it worked for a while, and now it has been broken for the past couple of months. Very strange!

**Comment by kburtram** [comment]
@benrr101 do you know if have any logging that would diagnosis this issue?

**Comment by weaslevdmd** [comment]
I am also seeing this starting today. I can force a connection to another server, modify the code and re-save, then execute the query again and sometimes I get the expected results, other times I get 0 records returned. At other times I get a blank result window similar to what jjcapron shows (but without the spinning Execute). If I attempt to cancel the query I get a message that no queries are running.

I am running on MacOS = High Sierra (Version 10.13.3)
VSCode = 1.21.0 (1.21.0)
MSSQL Extension = 1.3.0

This may actually be two separate issues (one for the 0 records returned, one for queries that don't execute correctly). I think one is a connection issue, the 0 records returned may be some kind of client or server caching problem.

**Comment by kburtram** [comment]
@weaslevdmd when a USE statement is encountered in a script the query execution engine will establish a new connection to the database referenced in the statement.  Do you happen to know if the USE statements were referencing the same DB that the editor was already connected to or a different DB?

**Comment by LouisDK** [comment]
I had the same issue - which was caused by having another SQL extension installed.
In my case the extension was *SQLTools* by Matheus Teixeira.
https://marketplace.visualstudio.com/items?itemName=mtxr.sqltools 

I uninstalled that extension, restarted VSCode, and queries started executing as expected.

**Comment by mainxx** [comment]
我也遇到同样的问题。
![3333](https://user-images.githubusercontent.com/14832636/50334861-8e9bc900-0544-11e9-83ef-f8693354b2e8.gif)

**Comment by antmarko** [comment]
I have experienced thee same issue. Has anyone found any solution?

**Comment by apoorvagarwal41** [comment]
I have the same issue, did anyone find a fix .

**Comment by apoorvagarwal41** [comment]
> I have the same issue, did anyone find a fix .

I found something which might help, the extension works totally fine while running vscode in windows, but hangs when vscode is running in a remote WSL environment.

**Comment by weaslevdmd** [comment]
@kburtram Sorry for the LONG delay in getting back to you, somehow the github thread got lost in my email.

Short answer is, yes, I believe the code I was using at the time had a USE statement that referenced another DB (perhaps on the same logical server) but was NOT the one the UI was currently connected to (i.e. the one shown in the bottom right when connected). The schema for all DB's involved were the same (DEV, TEST, PROD, TRAIN, etc. running the same DB / application).

I hope that helps.

P.S. I can also get this result with a different error message when I have mismatched comments. If I have nested /* --*/ comment blocks that are not perfectly paired I have issues too - but I think that is a different problem. See: https://github.com/microsoft/vscode-mssql/issues/1014
